src/Core/System/Snippet/Command/Util/CountryAgnosticFileValidator.php

```diff
@@ -0,0 +1,175 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\System\Snippet\Command\Util;
+
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\System\Snippet\SnippetException;
+use Shopware\Core\System\Snippet\SnippetPatterns;
+use Shopware\Core\System\Snippet\Struct\TranslationFile;
+use Shopware\Core\System\Snippet\Struct\TranslationFileCollection;
+use Shopware\Core\System\Snippet\Struct\ValidatedTranslationFileOptions;
+use Shopware\Core\System\Snippet\Struct\ValidatedTranslationFileStruct;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+
+/**
+ * @internal
+ */
+#[Package('discovery')]
+class CountryAgnosticFileValidator
+{
+    public const CORE_DOMAINS = [
+        'administration' => 'Administration',
+        'messages' => 'Core',
+        'storefront' => 'Storefront',
+    ];
+
+    public function getFinder(ValidatedTranslationFileOptions $options): Finder
+    {
+        $finder = (new Finder())
+            ->files()
+            ->ignoreUnreadableDirs()
+            ->ignoreDotFiles(true)
+            ->ignoreVCS(true)
+            ->exclude([
+                'node_modules',
+                'vendor',
+                'bin',
+                'static',
+                // Translations of languages fetched from crowdin should not be validated
+                'SwagLanguagePack/src/Resources/snippet',
+                'SwagLanguagePack/src/Resources/app/administration/src/snippet',
+                ...$options->getIgnoredPaths(),
+            ])
+            ->name([SnippetPatterns::CORE_SNIPPET_FILE_PATTERN, SnippetPatterns::ADMIN_SNIPPET_FILE_PATTERN])
+            ->sortByName(true);
+
+        if ($options->getDir()) {
+            $finder->in($options->getDir());
+        } elseif (empty($options->getExtensionPaths())) {
+            $finder->in('src');
+
+            if ($options->isAll()) {
+                $finder->in('custom/plugins');
```

Comment:
what about static-plugins?

Reply:
what about extensions located in vendor directory?

Reply:
I didn't know static-plugins exist. How can this happen?

if I allow vendor this tool would be unusable. But I'm open for suggestions how this can be handled in a better way.

Reply:
see in the other thread. you should use the path property of the extensions

Reply:
https://github.com/shopware/template/tree/trunk/custom/static-plugins